c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1044,6 +1050,21 @@ c2_status_t MfxC2DecoderComponent::QueryParam(const mfxVideoParam* src, C2Param:
                 }
                 break;
             }
+            case C2PortSurfaceAllocatorTuning::output::PARAM_TYPE: {
+                if (nullptr == *dst) {
+                    std::unique_ptr<C2PortSurfaceAllocatorTuning::output> outAlloc =
+                        std::make_unique<C2PortSurfaceAllocatorTuning::output>();
+
+                    outAlloc->value = C2PlatformAllocatorStore::BUFFERQUEUE;
```

Comment:
So this is used to override the default allocator GRALLOC? any improvement for Bufferqueue over GRALLOC?

Reply:
There is no performance gain for this change. It is just to return the default allocator when surface is being used. 
https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/frameworks/av/media/codec2/sfplugin/CCodecBufferChannel.cpp#1248

Reply:
OK, Thanks